### PR TITLE
Added filter for blog post feature media

### LINKED ISSUE #2516: Need filter(In code) for the blog post's feature image in the activity feed
###Scope
We need some filter for the blog post's feature image in the activity feed. Because we need to remove this filter from APP side. 
App Developers are facing some issue with this images so that why we remove image from API repose and pass it as separate object.

Please check video for the more details.
https://screencast-o-matic.com/watch/cr1FhbV1lcZ

Dependent on - https://github.com/buddyboss/buddyboss-app-native/issues/1093
https://github.com/buddyboss/buddyboss-platform-api/pull/268

###Acceptance Criteria
We need filter in below code. So, we can remove this filter in APP side.
add_filter( 'bp_get_activity_content_body', 'bp_blogs_activity_content_with_read_more', 9999, 2 );
image

###QA Needed

1. You need to check first with release branch and same testcases check with this PR. ( Not an issue just moved code in filter )
2. Settings -> Activity -> Post Type Comments -> Enable (Also select custom post type which you want to display in activity feed)
3. Create a diff blog post with long and short content and check in the activity feed
4. Just check ( I have just moved code in filter, not any issue in platform)

###ProductBoard URL
Enter the URL of this feature in product board.

### PR DESCRIPTION
**Pull request by jitendrabanjara1991**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Added below filter in the code and move feature image code in this filter.
add_filter( 'bb_add_feature_image_blog_post_as_avtivity_content', 'bb_add_feature_image_blog_post_as_avtivity_content_callback', 10, 2 );

Fixes #2516.

### How to test the changes in this Pull Request:

1. You need to check first with release branch and same testcases check with this PR. ( Not an issue just moved code in filter )
2. Settings -> Activity -> Post Type Comments -> Enable (Also select custom post type which you want to display in activity feed)
3. Create a diff blog post with long and short content and check in the activity feed
4. Just check ( I have just moved code in filter, not any issue in platform )

### Proof Screenshots or Video

https://screencast-o-matic.com/watch/cr1F13V1l1a

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

File: src/bp-activity/bp-activity-filters.php [hunk 1]
```diff
@@ -2409,3 +2403,22 @@ function bb_check_is_activity_content_empty( $data ) {
...
+ * @param int    $blog_post_id
+ * 
+ * @return string $content
+ * 
+ * @since x.x.x
+ */
+function bb_add_feature_image_blog_post_as_avtivity_content_callback( $content, $blog_post_id ) {
+	$src = wp_get_attachment_image_src( get_post_thumbnail_id( $blog_post_id ), 'full', false );
+	if( isset( $src[0] ) ) {
+		$content .=sprintf( ' <img src="%s">', esc_url( $src[0] ) );
```
Comment (chetansatasiya) [inline src/bp-activity/bp-activity-filters.php:2421]:
```suggestion
		$content .= sprintf( ' <img src="%s">', esc_url( $src[0] ) );
```

File: src/bp-activity/bp-activity-filters.php [hunk 2]
```diff
@@ -2409,3 +2403,22 @@ function bb_check_is_activity_content_empty( $data ) {
...
+add_filter( 'bb_add_feature_image_blog_post_as_avtivity_content', 'bb_add_feature_image_blog_post_as_avtivity_content_callback', 10, 2 );
+/**
+ * Function will add feature image for blog post in the acitivy feed content.
+ * 
+ * @param string $content
+ * @param int    $blog_post_id
+ * 
+ * @return string $content
+ * 
+ * @since x.x.x
```
Comment (chetansatasiya) [inline src/bp-activity/bp-activity-filters.php:2416]:
```suggestion
 * @since 1.6.4
```

**Comment by jitendrabanjara1991** [comment]
@chetansatasiya 
Request changes are done.
Please review again.
Thanks.

**Review by chetansatasiya — APPROVED** [review]

File: src/bp-activity/bp-activity-filters.php [hunk 3]
```diff
@@ -2319,10 +2319,7 @@ function bp_blogs_activity_content_with_read_more( $content, $activity ) {
 					$content = strip_tags( preg_replace( '/<iframe.*?\/iframe>/i', '', $content ), '<a>' );
 					$content .= $iframe;
 				} else {
-					$src = wp_get_attachment_image_src( get_post_thumbnail_id( $blog_post->ID ), 'full', false );
-					if( isset( $src[0] ) ) {
-						$content .= sprintf( ' <img src="%s">', esc_url( $src[0] ) );
-					}
+					$content = apply_filters( 'bb_add_feature_image_blog_post_as_avtivity_content', $content, $blog_post->ID );
```
Comment (chetansatasiya) [inline src/bp-activity/bp-activity-filters.php:2322]:
```suggestion
					$content = apply_filters( 'bb_add_feature_image_blog_post_as_activity_content', $content, $blog_post->ID );
```

File: src/bp-activity/bp-activity-filters.php [hunk 4]
```diff
@@ -2332,10 +2329,7 @@ function bp_blogs_activity_content_with_read_more( $content, $activity ) {
 				if( isset( $matches ) && array_key_exists( 0, $matches ) && ! empty( $matches[0] ) ) {
 					$content = $content;
 				} else {
-					$src = wp_get_attachment_image_src( get_post_thumbnail_id( $blog_post->ID ), 'full', false );
-					if( isset( $src[0] ) ) {
-						$content .= sprintf( ' <img src="%s">', esc_url( $src[0] ) );
-					}
+					$content = apply_filters( 'bb_add_feature_image_blog_post_as_avtivity_content', $content, $blog_post->ID );
```
Comment (chetansatasiya) [inline src/bp-activity/bp-activity-filters.php:2332]:
```suggestion
					$content = apply_filters( 'bb_add_feature_image_blog_post_as_activity_content', $content, $blog_post->ID );
```

File: src/bp-activity/bp-activity-filters.php [hunk 5]
```diff
@@ -2409,3 +2403,22 @@ function bb_check_is_activity_content_empty( $data ) {
 		return false;
 	}
 }
+
+add_filter( 'bb_add_feature_image_blog_post_as_avtivity_content', 'bb_add_feature_image_blog_post_as_avtivity_content_callback', 10, 2 );
```
Comment (chetansatasiya) [inline src/bp-activity/bp-activity-filters.php:2407]:
```suggestion
add_filter( 'bb_add_feature_image_blog_post_as_avtivity_content', 'bb_add_feature_image_blog_post_as_activity_content_callback', 10, 2 );
```

Comment (chetansatasiya) [inline src/bp-activity/bp-activity-filters.php:2407]:
```suggestion
add_filter( 'bb_add_feature_image_blog_post_as_avtivity_content', 'bb_add_feature_image_blog_post_as_activity_content_callback', 10, 2 );
```

File: src/bp-activity/bp-activity-filters.php [hunk 6]
```diff
@@ -2409,3 +2403,22 @@ function bb_check_is_activity_content_empty( $data ) {
...
+ * Function will add feature image for blog post in the activity feed content.
+ * 
+ * @param string $content
+ * @param int    $blog_post_id
+ * 
+ * @return string $content
+ * 
+ * @since 1.6.4
+ */
+function bb_add_feature_image_blog_post_as_avtivity_content_callback( $content, $blog_post_id ) {
```
Comment (chetansatasiya) [inline src/bp-activity/bp-activity-filters.php:2418]:
```suggestion
function bb_add_feature_image_blog_post_as_activity_content_callback( $content, $blog_post_id ) {
```

**Review by chetansatasiya — CHANGES_REQUESTED** [review]
@jitendrabanjara1991 small typo change

**Review by chetansatasiya — APPROVED** [review]

**Comment by jitendrabanjara1991** [comment]
@chetansatasiya 
cc - @KartikSuthar 
Request changes are done.
Please review again.
Thanks.

**Review by chetansatasiya — APPROVED** [review]

**Pull request merged**